packages/itwin/grouping-mapping-widget/src/widget/components/Grouping.tsx

```diff
@@ -314,6 +314,8 @@ export const Groupings = ({ mapping, goBack }: GroupsTreeProps) => {
     instance?: TableInstance,
   ): TableState => {
     switch (action.type) {
+      case "singleRowSelected":
+        action.value = true;
```

Comment:
Why do you need to rely on `action.value`? The state should have the info you need

Reply:
I was trying to be consistent. I was seeing singleRowSelected as more of a toggle as well. It wasn't too obvious for me that the row selected was passed in through selectedRowIds because the pattern was selectedRowIds was populated by all selected rows.